src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp

```diff
@@ -1047,14 +1041,10 @@ namespace
                 //  - throughput of the entire path up to the sampled point.
                 // Reference: "Practical and Controllable Subsurface Scattering
                 // for Production Path Tracing", p. 1 [ACM 2016 Article].
-                Spectrum precomputed_mis_weights;
-                if (channel_count == scattering_coef.size())
-                    precomputed_mis_weights = scattering_coef;
-                else
-                    Spectrum::upgrade(scattering_coef, precomputed_mis_weights);
+                Spectrum precomputed_mis_weights = scattering_coef;
                 precomputed_mis_weights *= vertex.m_throughput;
                 float max_density = 0.0f;
-                for (size_t i = 0; i < channel_count; ++i)
+                for (size_t i = 0; i < Spectrum::size(); ++i)
```

Comment:
I was doing that at first, then reverted to this simpler form... But after more thinking, it appears that maybe you're right and that I should change it everywhere. I'll check if there's any difference in the generated code.